### DISCUSSION TITLE
More aggressive yarn caching

### DISCUSSION THREAD

**Discussion by colophonemes** [vercel/community#164] [Help] [answered]
My builds are pretty slow, with the majority of time taken by `yarn install`. As far as I can tell, if `yarn.lock` changes at all, then the Yarn cache isn't restored.

Here's an excerpt from my build logs.

```
16:12:54.278  	Analyzing source code...
16:12:56.003  	Installing build runtime...
16:12:59.123  	Build runtime installed: 3.120s
16:13:02.735  	Looking up build cache...
16:13:03.140  	Build cache found. Downloading...
16:13:03.813  	Build cache downloaded [3.74 MB]: 672.959ms
16:13:05.263  	Installing dependencies...
16:13:05.555  	yarn install v1.22.10
16:13:05.712  	[1/5] Validating package.json...
16:13:05.712  	[2/5] Resolving packages...
16:13:06.665  	[3/5] Fetching packages...
16:13:49.500  	[4/5] Linking dependencies...
16:14:29.197  	[5/5] Building fresh packages...
16:14:32.684  	success Saved lockfile.
16:14:32.688  	Done in 87.14s.
```
As you can see, the total build cache is only 3.74MB, but most dependencies haven't changed. I'd expect that it'd still be beneficial to restore the previous build's Yarn cache, even if the lockfile has changed. Then Yarn can get most of the dependencies from the cache, and only needs to go over the network for anything not covered by the cache. For example, when I build on Heroku, it seems to restore the last know good build's cache, which means that the `yarn install` time is typically < 5s.

So, is it possible to opt in to using a previous build's Yarn cache even if the lockfile has changed?

**Comment by nk1tz** [comment]
bump

**Comment by Pieparker** [comment, marked as answer]
This conversation appears to have gone quiet. To keep discussions on topic and to prevent the resurrection of old topics, this discussion has been closed. If issues persist or a future reader stumbles upon this discussion and is exhibiting the same problem, please create a new discussion.
